UnityDebug/UnityAttach.cs

```diff
@@ -1,42 +1,46 @@
-﻿using System;
-using MonoDevelop.Debugger.Soft.Unity;
+﻿using MonoDevelop.Debugger.Soft.Unity;
 using System.Collections.Generic;
 using System.Linq;
 
 namespace UnityDebug
 {
-	public static class UnityAttach
-	{
-		static readonly Dictionary<string, string> targetNameToProcessName = new Dictionary<string,string> 
-		{
-			{ "unity editor", "Unity Editor" },
-			{ "osx player", "OSXPlayer" },
-			{ "windows player", "WindowsPlayer" },
-			{ "linux player", "WindowsPlayer" },
-			{ "ios player", "iPhonePlayer" },
-			{ "android player", "AndroidPlayer" }
-		};
-
-		public static IEnumerable<UnityProcessInfo> GetAttachableProcesses (string targetName)
-		{
-			string processName;
-
-			if (!targetNameToProcessName.TryGetValue (targetName.ToLower (), out processName))
-				return null;
-
-			UnityProcessDiscovery.GetProcessOptions options = UnityProcessDiscovery.GetProcessOptions.All;
-
-			if (processName == "Unity Editor")
-				options = UnityProcessDiscovery.GetProcessOptions.Editor;
-			else
-				options = UnityProcessDiscovery.GetProcessOptions.Players;
-
-			var processes = UnityProcessDiscovery.GetAttachableProcesses (options);
-
-			processes.ForEach(p => Log.Write("Found Unity process: " + p.Name + " (" + p.Id + ")"));
-
-			return processes.Where (p => p.Name.Contains (processName));
-		}
-	}
+    public static class UnityAttach
+    {
+        static readonly Dictionary<string, string> targetNameToProcessName = new Dictionary<string, string>
+        {
+            { "unity editor", "Unity Editor" },
+            { "osx player", "OSXPlayer" },
+            { "windows player", "WindowsPlayer" },
+            { "linux player", "WindowsPlayer" },
+            { "ios player", "iPhonePlayer" },
+            { "android player", "AndroidPlayer" }
+        };
+
+
+        public static IEnumerable<UnityProcessInfo> GetAttachableProcesses(string targetName)
+        {
+            string processName;
+
+            UnityProcessDiscovery.GetProcessOptions options = UnityProcessDiscovery.GetProcessOptions.All;
+
+            if (!targetNameToProcessName.TryGetValue(targetName.ToLower(), out processName))
+            {
+                processName = targetName;
```

Comment:
Yes. Since the old version could only allow you to connect to Unity Instances that matched values within targetNameToProcessName. Since there could be multiple similar Unity instances detected, this change allows for the old functionality and defaults to continue working, and still allow for non standard target names.